lib/jobs/obm-control.js

```diff
@@ -64,18 +62,31 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
-                return setting ? setting.service === self.obmServiceName : false;
-            });
-
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                self.settings = _.find(node.obmSettings, function(setting) {
+                    return setting ? setting.service === self.obmServiceName : false;
+                });
+                // throw error if the exact obm service doesn't exist
+                assert.ok(self.settings, "Node should have OBM settings for service: " +
+                    self.obmServiceName);
+            } else {
+                //find the default obm service
+                if(node.obmSettings.length === 1) {
+                    self.settings = node.obmSettings[0];
+                    logger.info("Defaulting to obmServiceName "+ self.settings.service +
+                        " for node because only one OBM service present.");
+                }else {
+                    assert.fail(true,true, "More than one OBM service assigned to this node," +
+                        " (no default set for which to use)" +
```

Comment:
So if there is no obm settings defined, it actually is caught on line 63... this last case was a way for me to satisfy Felix's request to handle the edge case of having the obm property exist as an empty array... he recommended just wrapping them into a "catch all" error, and I choose it to simplify unit testing and code coverage. Should I break out this logic again and reimplement a unique error? @benbp @yyscamper